<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Predict cell type proportion based on DNA methylation data • CellsPickMe</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="lightswitch.js"></script><script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Predict cell type proportion based on DNA methylation data">
<meta name="description" content="The package streamline cell type prediction process, from reference selection, machine-learning algorithm-based feature selection, to cell type prediction, and includes methods to assess prediction accuracy based on clustering and rmse calculation.">
<meta property="og:description" content="The package streamline cell type prediction process, from reference selection, machine-learning algorithm-based feature selection, to cell type prediction, and includes methods to assess prediction accuracy based on clustering and rmse calculation.">
<meta property="og:image" content="https://maggie-fu.github.io/CellsPickMe/logo.png">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">CellsPickMe</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/CellsPickMe.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="articles/UniBlood_references.html">UniBlood Reference Creation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="cellspickme">CellsPickMe<a class="anchor" aria-label="anchor" href="#cellspickme"></a>
</h1>
</div>
<!-- badges: start -->

<!-- badges: end -->
<p><img src="logo2.png" width="25%" height="25%"></p>
<p>The <strong>CellsPickMe</strong> package takes DNA methylation (DNAme) data generated from Illumina microarray of a heterogeneous tissue and predict its cellular composition based on cell types available in the reference profiles. <em>Notably, for each population of <strong>cells</strong>, the package “<strong>picks</strong>” DNA<strong>me</strong> features that best predict cellular identities with machine learning algorithms to improve deconvolution performance.</em> Currently, the algorithm is compatible for peripheral blood, cord blood, saliva, and brain (neuron vs non-neuron).</p>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of CellsPickMe from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"maggie-fu/CellsPickMe"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>Reference-based cell type deconvolution includes four steps: reference dataset selection, data normalization, feature selection, and cell composition regression.</p>
<div class="section level3">
<h3 id="id_1-obtain-reference-dataset">1. Obtain reference dataset<a class="anchor" aria-label="anchor" href="#id_1-obtain-reference-dataset"></a>
</h3>
<p>From the available list of reference datasets (“Reinius”, “IDOL”, “Extended”, “UniBlood7”, “UniBlood13”, “UniBlood19”, “Cord”, “DLPFC”, and “Middleton”), select one that is appropriate for your sample (based on tissue and sample age) For more detail regarding the references, check out the <a href="https://maggie-fu.github.io/CellsPickMe/articles/CellsPickMe.html">vignette</a> and <a href="https://maggie-fu.github.io/CellsPickMe/articles/UniBlood_references.html">UniBlood Reference Generation</a> page</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://maggie-fu.github.io/CellsPickMe/">CellsPickMe</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Request the IDOL reference (2016) with no normalization</span></span>
<span><span class="va">ref_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/getRef.html">getRef</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="st">"UniBlood19"</span>, normType <span class="op">=</span> <span class="st">"None"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_2-normalize-sample-and-reference-datasets-together">2. Normalize sample and reference datasets together<a class="anchor" aria-label="anchor" href="#id_2-normalize-sample-and-reference-datasets-together"></a>
</h3>
<p>Normalize user’s sample and reference data sets together to reduce batch effect and improve prediction accuracy. Option for <code>normType</code> includes “Noob”, “Funnorm”, “Quantile”, “Quantile.b”, “None”, with the first 3 options being exclusively for RGChannelSet objects, and Quantile.b for beta matrix.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example blood cell mixture, subsetted from the IDOL dataset (GSE110554)</span></span>
<span><span class="va">test_dat</span> <span class="op">&lt;-</span> <span class="fu">CellsPickMe</span><span class="fu">::</span><span class="va"><a href="reference/IDOL_mixed_cells.html">IDOL_mixed_cells</a></span></span>
<span>    </span>
<span><span class="co"># Combine sample and reference data sets together, followed by normalization (if selected)</span></span>
<span><span class="va">comb_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/combData.html">combData</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">test_dat</span>, </span>
<span>                     reference <span class="op">=</span> <span class="va">ref_dat</span><span class="op">$</span><span class="va">reference</span>,</span>
<span>                     cellTypes <span class="op">=</span> <span class="va">ref_dat</span><span class="op">$</span><span class="va">cellTypes</span>, </span>
<span>                     class <span class="op">=</span> <span class="st">"rgset"</span>,               <span class="co">#c("rgset", "betas")</span></span>
<span>                     normType <span class="op">=</span> <span class="st">"Quantile"</span><span class="op">)</span>         <span class="co"># c("Noob", "Funnorm", "Quantile", "Quantile.b", "None")</span></span>
<span><span class="co">#&gt; Combining Data with Flow Sorted Data and Normalizing.</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="id_3-pick-features-that-best-distinguish-cell-type">3. Pick features that best distinguish cell type<a class="anchor" aria-label="anchor" href="#id_3-pick-features-that-best-distinguish-cell-type"></a>
</h3>
<p>The <strong>CellsPickMe</strong> package supports feature selection with either the traditional T-test, or with machine-learning-based methods such as elastic net and random forest to obtain a curated list of features that are highly predictive of cell types</p>
<p>Here is the T-test based feature selection method.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pick probes with T tests</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pickProbes.html">pickProbes</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                     probeList <span class="op">=</span> <span class="st">"Ttest"</span>, <span class="co">#c("Ttest", "IDOL")</span></span>
<span>                     probeSelect <span class="op">=</span> <span class="st">"both"</span>, <span class="co">#c("both", "any", "pval")</span></span>
<span>                     nProbes <span class="op">=</span> <span class="fl">100</span>, <span class="co"># number of probes to pick for each cell type</span></span>
<span>                     p.val <span class="op">=</span> <span class="fl">0.05</span>,  <span class="co"># max pval</span></span>
<span>                     min.delta.beta <span class="op">=</span> <span class="fl">0.05</span>, <span class="co"># min delta beta</span></span>
<span>                     plotRef <span class="op">=</span> <span class="cn">T</span>, <span class="co"># plot heatmap?</span></span>
<span>                     verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span><span class="co">#&gt; Estimating Weights for Cell Type Prediction Based on Selected Probeset.</span></span></code></pre></div>
<p><img src="reference/figures/README-pickProbes1-1.png" width="100%"></p>
<p>Alternatively, here are some options for machine-learning-based feature selection.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">### Set up server for parallelization - run the code if picking probes with Caret</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RevolutionAnalytics/doparallel" class="external-link">doParallel</a></span><span class="op">)</span></span>
<span><span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu">makeCluster</span><span class="op">(</span><span class="fu">detectCores</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># change as needed</span></span>
<span><span class="fu">registerDoParallel</span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pick probes with repeated cross validation with lasso and elastic net (EN)</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/pickProbes.html">pickProbes</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                     probeList <span class="op">=</span> <span class="st">"Caret_CV"</span>, <span class="co">#c("Caret_CV", "Caret_LOOCV")</span></span>
<span>                     caretMods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"RF"</span>, <span class="st">"EN"</span><span class="op">)</span>,  <span class="co">#c("lasso", "EN", "BLR", "CART", "RF", "GBM", "PLDA", "GAnRF", "GAnNB", "GAnSVM", "GAnNN")</span></span>
<span>                     probeSelect <span class="op">=</span> <span class="st">"any"</span>,</span>
<span>                     p.val <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     min.delta.beta <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                     filterK <span class="op">=</span> <span class="fl">1000</span>, <span class="co"># number of probes to put into the predictor for each cell type</span></span>
<span>                     seed <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     plotRef <span class="op">=</span> <span class="cn">F</span>, <span class="co"># plot heatmap?</span></span>
<span>                     verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Picked probes with RF and the estimated coefficients</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">$</span><span class="va">coefs</span><span class="op">$</span><span class="va">probeCoefs</span><span class="op">$</span><span class="va">RF</span><span class="op">)</span></span>
<span><span class="co">#&gt;            Bcell_cord      Bmem       Bnv CD8T_cord    CD8mem     CD8nv</span></span>
<span><span class="co">#&gt; cg24262870  0.1235089 0.7885589 0.1454030 0.7988769 0.9002228 0.8815243</span></span>
<span><span class="co">#&gt; cg24667756  0.1468955 0.3246588 0.1774464 0.8213872 0.8656461 0.8369870</span></span>
<span><span class="co">#&gt; cg21531416  0.1954677 0.7161480 0.2611708 0.6352860 0.9123706 0.8243531</span></span>
<span><span class="co">#&gt; cg27436324  0.2643973 0.7655840 0.2042478 0.8774527 0.9099452 0.8940336</span></span>
<span><span class="co">#&gt; cg18406792  0.1745236 0.4272825 0.1660204 0.8060743 0.8471631 0.8077735</span></span>
<span><span class="co">#&gt; cg15198193  0.1576836 0.7751659 0.4488992 0.6089112 0.8850735 0.6163415</span></span>
<span><span class="co">#&gt;            CD4T_cord    CD4mem     CD4nv      Treg Gran_cord       Bas</span></span>
<span><span class="co">#&gt; cg24262870 0.6986202 0.8848730 0.8628457 0.8774487 0.8433450 0.8428693</span></span>
<span><span class="co">#&gt; cg24667756 0.8015645 0.8815034 0.8414196 0.8673486 0.8545976 0.8684859</span></span>
<span><span class="co">#&gt; cg21531416 0.8386425 0.9126800 0.8887243 0.9108057 0.7946011 0.8733178</span></span>
<span><span class="co">#&gt; cg27436324 0.8877283 0.9060499 0.8918332 0.9088491 0.9053751 0.9017803</span></span>
<span><span class="co">#&gt; cg18406792 0.8015001 0.8518648 0.8016774 0.8430231 0.8296527 0.8415291</span></span>
<span><span class="co">#&gt; cg15198193 0.4571046 0.8280606 0.4386012 0.8255160 0.8851114 0.8909355</span></span>
<span><span class="co">#&gt;                  Eos       Neu Mono_cord      Mono   NK_cord        NK</span></span>
<span><span class="co">#&gt; cg24262870 0.8384348 0.8475652 0.8450575 0.8577191 0.8034451 0.8599368</span></span>
<span><span class="co">#&gt; cg24667756 0.8734111 0.8663822 0.8567890 0.8722139 0.7773920 0.8388949</span></span>
<span><span class="co">#&gt; cg21531416 0.8870945 0.8667716 0.7915671 0.8902495 0.8265656 0.8828000</span></span>
<span><span class="co">#&gt; cg27436324 0.9044173 0.9059902 0.9076917 0.9014642 0.8858523 0.8931943</span></span>
<span><span class="co">#&gt; cg18406792 0.8097433 0.8522817 0.8409429 0.8522421 0.7989837 0.8274087</span></span>
<span><span class="co">#&gt; cg15198193 0.8939239 0.9035070 0.8632832 0.8930117 0.7029587 0.7756338</span></span>
<span><span class="co">#&gt;                 nRBC</span></span>
<span><span class="co">#&gt; cg24262870 0.7149653</span></span>
<span><span class="co">#&gt; cg24667756 0.8298977</span></span>
<span><span class="co">#&gt; cg21531416 0.8788950</span></span>
<span><span class="co">#&gt; cg27436324 0.9048735</span></span>
<span><span class="co">#&gt; cg18406792 0.7034571</span></span>
<span><span class="co">#&gt; cg15198193 0.8892714</span></span></code></pre></div>
<div class="section level4">
<h4 id="assess-clustering-stability">Assess clustering stability<a class="anchor" aria-label="anchor" href="#assess-clustering-stability"></a>
</h4>
<p>To further evaluate the performance of the selected probes, pvClust can be applied to assess whether the picked probes can be used to generate the correct cluster (cell type labeling) in reference data</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clustAU</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/identClust.html">identClust</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>,</span>
<span>                      probes <span class="op">=</span> <span class="va">probes</span>,</span>
<span>                      parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span><span class="co">#&gt; Creating a temporary cluster...done:</span></span>
<span><span class="co">#&gt; socket cluster with 15 nodes on host 'localhost'</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span>
<span><span class="co">#&gt; Creating a temporary cluster...done:</span></span>
<span><span class="co">#&gt; socket cluster with 15 nodes on host 'localhost'</span></span>
<span><span class="co">#&gt; Multiscale bootstrap... Done.</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="id_4-estimate-cell-type-proportion">4. Estimate cell type proportion<a class="anchor" aria-label="anchor" href="#id_4-estimate-cell-type-proportion"></a>
</h3>
<p>Finally, the selected features are used to estimate cell type proportions in the sample data set. We also incorporate the CETYGO score (see Reference) to estimate prediction performance even in the absence of a validation cohort / ground truth cell count.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="reference/predictCT.html">predictCT</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                 probes <span class="op">=</span> <span class="va">probes</span>, </span>
<span>                 method <span class="op">=</span> <span class="st">"CP"</span>,  <span class="co">#c("CP", "RPC", "SVR")</span></span>
<span>                 removenRBC <span class="op">=</span> <span class="cn">F</span>, <span class="co"># remove nRBC?</span></span>
<span>                 verbose <span class="op">=</span> <span class="cn">F</span>, </span>
<span>                 cetygo <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="co"># CETYGO to assess reference appropriateness (RMSE evaluation)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="examine-output">Examine output<a class="anchor" aria-label="anchor" href="#examine-output"></a>
</h3>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimated proportions based on UniBlood19 reference, Quantile normalization, RF feature selection, and constraint projection</span></span>
<span><span class="co"># CETYOGO score column shows the CETYGO score for each sample</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">RF</span><span class="op">)</span></span>
<span><span class="co">#&gt;                        Bcell_cord        Bmem          Bnv     CD8T_cord</span></span>
<span><span class="co">#&gt; 201868590193_R01C01  4.900574e-03 0.033452481 1.379462e-01  1.514549e-17</span></span>
<span><span class="co">#&gt; 201868590243_R02C01  3.651905e-18 0.003854140 2.263836e-02  1.437579e-17</span></span>
<span><span class="co">#&gt; 201868590267_R01C01  8.517287e-19 0.001492958 7.777743e-03  9.612166e-18</span></span>
<span><span class="co">#&gt; 201868590267_R05C01 -4.750408e-19 0.000000000 8.474686e-19 -5.644007e-18</span></span>
<span><span class="co">#&gt; 201869680008_R01C01 -8.604787e-18 0.016948846 4.167371e-02 -2.885869e-18</span></span>
<span><span class="co">#&gt; 201869680008_R03C01  4.207191e-18 0.001690919 1.916507e-04  1.366324e-17</span></span>
<span><span class="co">#&gt;                         CD8mem      CD8nv   CD4T_cord     CD4mem        CD4nv</span></span>
<span><span class="co">#&gt; 201868590193_R01C01 0.09173997 0.10524220 0.003926924 0.03872764 0.0000000000</span></span>
<span><span class="co">#&gt; 201868590243_R02C01 0.01850916 0.05208359 0.029338138 0.07047455 0.0200054165</span></span>
<span><span class="co">#&gt; 201868590267_R01C01 0.02684480 0.05160417 0.016167004 0.05741796 0.0005965834</span></span>
<span><span class="co">#&gt; 201868590267_R05C01 0.04888686 0.09128600 0.031025712 0.09102361 0.0067783757</span></span>
<span><span class="co">#&gt; 201869680008_R01C01 0.12561960 0.16432521 0.033519080 0.08303561 0.0020908658</span></span>
<span><span class="co">#&gt; 201869680008_R03C01 0.03353352 0.07468721 0.021948900 0.07435698 0.0093463515</span></span>
<span><span class="co">#&gt;                            Treg     Gran_cord           Bas           Eos</span></span>
<span><span class="co">#&gt; 201868590193_R01C01 0.001226282 -9.632261e-18  1.724723e-18 -3.909634e-18</span></span>
<span><span class="co">#&gt; 201868590243_R02C01 0.019797856 -2.398769e-17  1.834631e-18  0.000000e+00</span></span>
<span><span class="co">#&gt; 201868590267_R01C01 0.008459640 -1.141656e-17 -2.596394e-18  1.631795e-18</span></span>
<span><span class="co">#&gt; 201868590267_R05C01 0.014504984 -3.927637e-17  1.718343e-18  1.459127e-19</span></span>
<span><span class="co">#&gt; 201869680008_R01C01 0.015657041 -3.078488e-17 -7.265285e-19  0.000000e+00</span></span>
<span><span class="co">#&gt; 201869680008_R03C01 0.015005326 -1.206880e-17  2.004191e-18  0.000000e+00</span></span>
<span><span class="co">#&gt;                           Neu     Mono_cord       Mono       NK_cord</span></span>
<span><span class="co">#&gt; 201868590193_R01C01 0.2342231  2.738654e-19 0.20168628 -1.682621e-17</span></span>
<span><span class="co">#&gt; 201868590243_R02C01 0.7117166  5.758737e-18 0.04418971 -1.019774e-17</span></span>
<span><span class="co">#&gt; 201868590267_R01C01 0.7336139 -2.737916e-18 0.09906315 -1.600910e-18</span></span>
<span><span class="co">#&gt; 201868590267_R05C01 0.6399841 -1.116955e-18 0.06725238  2.389558e-17</span></span>
<span><span class="co">#&gt; 201869680008_R01C01 0.1218932  1.143990e-18 0.24001386 -8.338604e-18</span></span>
<span><span class="co">#&gt; 201869680008_R03C01 0.6862866  7.171603e-18 0.05876430 -1.070111e-17</span></span>
<span><span class="co">#&gt;                               NK          nRBC     CETYGO</span></span>
<span><span class="co">#&gt; 201868590193_R01C01 1.498043e-01  4.923736e-19 0.03460435</span></span>
<span><span class="co">#&gt; 201868590243_R02C01 8.779442e-03 -7.450351e-19 0.03319760</span></span>
<span><span class="co">#&gt; 201868590267_R01C01 2.421607e-18  8.673617e-19 0.03196155</span></span>
<span><span class="co">#&gt; 201868590267_R05C01 1.026174e-02 -1.487835e-18 0.03414952</span></span>
<span><span class="co">#&gt; 201869680008_R01C01 1.568069e-01 -1.228120e-18 0.03585871</span></span>
<span><span class="co">#&gt; 201869680008_R03C01 2.326621e-02 -8.343516e-19 0.03310038</span></span>
<span></span>
<span><span class="co"># Basic visualization of deconvolution output</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">RF</span><span class="op">[</span>, <span class="va">ref_dat</span><span class="op">$</span><span class="va">cellTypes</span><span class="op">]</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op"><a href="reference/pipe.html">%&gt;%</a></span> </span>
<span>    <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">variable</span>, <span class="va">value</span>, color <span class="op">=</span> <span class="va">variable</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"cell type"</span>, </span>
<span>         y <span class="op">=</span> <span class="st">"cell type proportion (%)"</span>, </span>
<span>         title <span class="op">=</span> <span class="st">"Blood cell type proportion"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Bcell_cord <span class="op">=</span> <span class="st">"#9C9EDEFF"</span>, </span>
<span>                                  Bnv <span class="op">=</span> <span class="st">"#7375B5FF"</span>, </span>
<span>                                  Bmem <span class="op">=</span> <span class="st">"#4A5584FF"</span>, </span>
<span>                                  Bcell <span class="op">=</span> <span class="st">"#7375B5FF"</span>,</span>
<span>                                  CD4T_cord <span class="op">=</span> <span class="st">"#CEDB9CFF"</span>, </span>
<span>                                  CD4nv <span class="op">=</span> <span class="st">"#B5CF6BFF"</span>, </span>
<span>                                  CD4mem <span class="op">=</span> <span class="st">"#637939FF"</span>, </span>
<span>                                  Treg <span class="op">=</span> <span class="st">"#8CA252FF"</span>, </span>
<span>                                  CD4T <span class="op">=</span> <span class="st">"#B5CF6BFF"</span>,</span>
<span>                                  CD8T_cord <span class="op">=</span> <span class="st">"#E7CB94FF"</span>, </span>
<span>                                  CD8nv <span class="op">=</span> <span class="st">"#E7BA52FF"</span>, </span>
<span>                                  CD8mem <span class="op">=</span> <span class="st">"#8C6D31FF"</span>, </span>
<span>                                  CD8T <span class="op">=</span> <span class="st">"#E7BA52FF"</span>,</span>
<span>                                  NK_cord <span class="op">=</span> <span class="st">"#7BBCB0FF"</span>, </span>
<span>                                  NK <span class="op">=</span> <span class="st">"#3A7C89FF"</span>, </span>
<span>                                  Mono_cord <span class="op">=</span> <span class="st">"#F3CBD3FF"</span>, </span>
<span>                                  Mono <span class="op">=</span> <span class="st">"#707070"</span>,</span>
<span>                                  Gran_cord <span class="op">=</span> <span class="st">"#D39C83FF"</span>, </span>
<span>                                  Gran <span class="op">=</span> <span class="st">"#A65461FF"</span>, </span>
<span>                                  Neu <span class="op">=</span> <span class="st">"#A65461FF"</span>, </span>
<span>                                  Bas <span class="op">=</span> <span class="st">"#7B4173FF"</span>,</span>
<span>                                  Eos <span class="op">=</span> <span class="st">"#A55194FF"</span>, </span>
<span>                                  nRBC <span class="op">=</span> <span class="st">"#843C39FF"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">0.9</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="reference/figures/README-visualization-1.png" width="100%"></p>
</div>
</div>
<div class="section level2">
<h2 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h2>
<p>The manuscript detailing CellsPickMe and its use is currently under preparation. For more information about this please contact Maggie Fu at <a href="mailto:maggie.fu@bcchr.ca" class="email">maggie.fu@bcchr.ca</a>.</p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>Depending on the options you used, please consider citing the following references as this package is built on their data / code / papers.</p>
<ol style="list-style-type: decimal">
<li>
<p>DS Vellame et al. (2023). Uncertainty quantification of reference-based cellular deconvolution algorithms. <em>Epigenetics</em> 18, 1: 2137659. doi: <a href="https://doi.org/10.1080/15592294.2022.2137659" class="external-link">10.1080/15592294.2022.2137659</a></p>
<ul>
<li>Please cite this paper if you set the parameter <code>cetygo = T</code> for <code><a href="reference/predictCT.html">predictCT()</a></code>
</li>
</ul>
</li>
<li>
<p>LE Reinius et al. (2012). Differential DNA methylation in purified human blood cells: implications for cell lineage and studies on disease susceptibility. <em>PloS one</em>. <em>7</em>(7), e41361. doi: <a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "Reinius")</code>
</li>
</ul>
</li>
<li>
<p>DC Koestler et al. (2016). Improving cell mixture deconvolution by identifying optimal DNA methylation libraries (IDOL). <em>BMC bioinformatics</em>. 17, 120. doi: <a href="https://doi.org/10.1186/s12859-016-0943-7" class="external-link">10.1186/s12859-016-0943-7</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "IDOL")</code>
</li>
</ul>
</li>
<li>
<p>LA Salas et al. (2022) Enhanced cell deconvolution of peripheral blood using DNA methylation for high-resolution immune profiling. <em>Nat Commun</em>. 13, 761. doi: <a href="https://doi.org/10.1038/s41467-021-27864-7" class="external-link">10.1038/s41467-021-27864-7</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "Extended")</code>
</li>
</ul>
</li>
<li>
<p>K Gervin et al. (2019). Systematic evaluation and validation of reference and library selection methods for deconvolution of cord blood DNA methylation data. <em>Clinical epigenetics</em>. 11, 1-15. doi: <a href="https://doi.org/10.1186/s13148-019-0717-y" class="external-link">10.1186/s13148-019-0717-y</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "Cord")</code>
</li>
</ul>
</li>
<li>
<p>LY Middleton et al. (2022). Saliva cell type DNA methylation reference panel for epidemiological studies in children. <em>Epigenetics</em>, <em>17</em>(2), 161-177. doi: <a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "Middleton")</code>
</li>
</ul>
</li>
<li>
<p>J Guintivano et al. (2013). A cell epigenotype specific model for the correction of brain cellular heterogeneity bias and its application to age, brain region and major depression. <em>Epigenetics</em>, <em>8</em>(3), 290-302. doi: <a href="https://doi.org/10.4161/epi.23924" class="external-link">10.4161/epi.23924</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset, i.e. <code>getRef(ref = "DLPFC")</code>
</li>
</ul>
</li>
<li>
<p>TJ Triche, et al. (2013). Low-level processing of Illumina Infinium DNA Methylation BeadArrays. <em>Nucleic Acids Res</em>. 41, e90. doi: <a href="http://www.dx.doi.org/10.1093/nar/gkt090" class="external-link">10.1093/nar/gkt090</a>.</p>
<ul>
<li>Please cite this paper if you used this normalization method, i.e. <code>getRef(normType = "Noob")</code> or <code>combData(normType = "Noob")</code>
</li>
</ul>
</li>
<li>
<p>JP Fortin et al. (2014). Functional normalization of 450k methylation array data improves replication in large cancer studies. <em>Genome Biology</em> 15, 503. doi: <a href="http://www.dx.doi.org/10.1186/s13059-014-0503-2" class="external-link">10.1186/s13059-014-0503-2</a>.</p>
<ul>
<li>Please cite this paper if you used this normalization method, i.e. <code>getRef(normType = "Funnorm")</code> or <code>combData(normType = "Funnorm")</code>
</li>
</ul>
</li>
<li>
<p>N Touleimat and J Tost. (2012). <em>Complete pipeline for Infinium Human Methylation 450K BeadChip data processing using subset quantile normalization for accurate DNA methylation estimation.</em> <em>Epigenomics</em> 4, 325-341. doi: <a href="https://doi.org/10.2217/epi.12.21" class="external-link">10.2217/epi.12.21</a></p>
<ul>
<li>Please cite this paper if you used this normalization method, i.e. <code>getRef(normType = "Quantile")</code> or <code>combData(normType = "Quantile")</code>
</li>
</ul>
</li>
</ol>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small>GPL (&gt;= 3)</small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing CellsPickMe</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Maggie Fu <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/YOUR-ORCID-ID" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>

<div class="dev-status">
<h2 data-toc-skip>Dev status</h2>
<ul class="list-unstyled">
<li><a href="https://doi.org/10.5281/zenodo.15225880" class="external-link"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.15225880.svg"></a></li>
</ul>
</div>

  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Maggie Fu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
