<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>CellsPickMe • CellsPickMe</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../lightswitch.js"></script><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="CellsPickMe">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top " aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">CellsPickMe</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/CellsPickMe.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/UniBlood_references.html">UniBlood Reference Creation</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-lightswitch" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true" aria-label="Light switch"><span class="fa fa-sun"></span></button>
  <ul class="dropdown-menu dropdown-menu-end" aria-labelledby="dropdown-lightswitch">
<li><button class="dropdown-item" data-bs-theme-value="light"><span class="fa fa-sun"></span> Light</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="dark"><span class="fa fa-moon"></span> Dark</button></li>
    <li><button class="dropdown-item" data-bs-theme-value="auto"><span class="fa fa-adjust"></span> Auto</button></li>
  </ul>
</li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>CellsPickMe</h1>
                        <h4 data-toc-skip class="author">Maggie Fu</h4>
            <address class="author_afil">
      The University of British
Columbia<br><a class="author_email" href="mailto:#"></a><a href="mailto:maggie.fu@bcchr.ca" class="email">maggie.fu@bcchr.ca</a>
      </address>
                  
      

      <div class="d-none name"><code>CellsPickMe.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h3>
<p><strong>CellsPickMe</strong> is a streamlined R package for cell type
proportion prediction in a heterogeneous tissue (i.e. deconvolution)
based on DNA methylation (DNAme) data. As cell identity is crucially
linked to cell function and DNAme profiles, estimation of cell type
proportion is instrumental to understanding major factors driving DNAme
variability. Accounting for inter-individual cellular heterogeneity is
also vital in epigenome-wide association studies to accurately evaluate
the signals associated with the variables of interest.</p>
<p>The <strong>CellsPickMe</strong> package takes DNAme data generated
from Illumina microarray and predict its cellular composition based on
cell types available in the reference profiles. Currently, the algorithm
is compatible for peripheral blood, cord blood, saliva, and brain
(neuron vs non-neuron). The table below illustrate the reference
datasets available in the <strong>CellsPickMe</strong> package for
deconvolution. We curated the UniBlood references (7, 13, and 19) to
address the current challenge in deconvoluting longitudinal and
pediatric blood samples. Refer to the [UniBlood Reference
Creation][UniBlood Reference Creation] article for details of the
UniBlood references.</p>
<div class="section level5">
<h5 id="available-reference-datasets-for-tissues-in-cellspickme">Available reference datasets for tissues in CellsPickMe<a class="anchor" aria-label="anchor" href="#available-reference-datasets-for-tissues-in-cellspickme"></a>
</h5>
<table style="width:99%;" class="table">
<colgroup>
<col width="9%">
<col width="6%">
<col width="8%">
<col width="74%">
</colgroup>
<thead><tr class="header">
<th>Tissue</th>
<th>Reference dataset</th>
<th>ID for the <code><a href="../reference/getRef.html">getRef()</a></code> function</th>
<th>DOI / Bioconductor Package</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Adult peripheral blood</td>
<td>Reinius (2012)</td>
<td>Reinius</td>
<td>
<p>LE Reinius et al. (2012). Differential DNA methylation in
purified human blood cells: implications for cell lineage and studies on
disease susceptibility. <em>PloS one</em>. <em>7</em>(7), e41361. doi:
<a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<pre><code><span><span class="va">FlowSorted.Blood.450k</span></span></code></pre>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.FlowSorted.Blood.450k" title="DOI for use in publications, etc., will always redirect to current release version (or devel if package is not in release yet)." class="external-link">10.18129/B9.bioc.FlowSorted.Blood.450k</a></p>
</td>
</tr>
<tr class="even">
<td>Adult peripheral blood</td>
<td>IDOL (2016)</td>
<td>IDOL</td>
<td>
<p>DC Koestler et al. (2016). Improving cell mixture deconvolution
by identifying optimal DNA methylation libraries (IDOL). <em>BMC
bioinformatics</em>. 17, 120. doi: <a href="https://doi.org/10.1186/s12859-016-0943-7" class="external-link">10.1186/s12859-016-0943-7</a></p>
<pre><code><span><span class="va">FlowSorted.Blood.EPIC</span></span></code></pre>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.FlowSorted.Blood.EPIC" title="DOI for use in publications, etc., will always redirect to current release version (or devel if package is not in release yet)." class="external-link">10.18129/B9.bioc.FlowSorted.Blood.EPIC</a></p>
</td>
</tr>
<tr class="odd">
<td>Adult peripheral blood</td>
<td>IDOL extended (2022)</td>
<td>Extended</td>
<td>
<p>LA Salas et al. (2022) Enhanced cell deconvolution of peripheral
blood using DNA methylation for high-resolution immune profiling.
<em>Nat Commun</em>. 13, 761. doi: <a href="https://doi.org/10.1038/s41467-021-27864-7" class="external-link">10.1038/s41467-021-27864-7</a></p>
<pre><code><span><span class="va">FlowSorted.BloodExtended.EPIC</span></span></code></pre>
<p><a href="https://doi.org/10.5281/zenodo.5256298" class="external-link">10.5281/zenodo.5256297</a></p>
</td>
</tr>
<tr class="even">
<td>Cord blood</td>
<td>Cord (2018)</td>
<td>Cord</td>
<td>
<p>K Gervin et al. (2019). Systematic evaluation and validation of
reference and library selection methods for deconvolution of cord blood
DNA methylation data. <em>Clinical epigenetics</em>. 11, 1-15. doi: <a href="https://doi.org/10.1186/s13148-019-0717-y" class="external-link">10.1186/s13148-019-0717-y</a></p>
<pre><code><span><span class="va">FlowSorted.CordBlood.450k</span></span></code></pre>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.FlowSorted.CordBlood.450k" title="DOI for use in publications, etc., will always redirect to current release version (or devel if package is not in release yet)." class="external-link">10.18129/B9.bioc.FlowSorted.CordBlood.450k</a></p>
</td>
</tr>
<tr class="odd">
<td>Cord and adult peripheral blood</td>
<td>UniBlood7</td>
<td>UniBlood7</td>
<td>Available soon</td>
</tr>
<tr class="even">
<td>Cord and adult peripheral blood</td>
<td>UniBlood13</td>
<td>UniBlood13</td>
<td>Available soon</td>
</tr>
<tr class="odd">
<td>Cord and adult peripheral blood</td>
<td>UniBlood19</td>
<td>UniBlood19</td>
<td>Available soon</td>
</tr>
<tr class="even">
<td>Saliva</td>
<td>Middleton (2022)</td>
<td>Middleton</td>
<td>
<p>LY Middleton et al. (2022). Saliva cell type DNA methylation
reference panel for epidemiological studies in children.
<em>Epigenetics</em>, <em>17</em>(2), 161-177. doi: <a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<pre><code><span><span class="va">BeadSorted.Saliva.EPIC</span></span></code></pre>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.BeadSorted.Saliva.EPIC" title="DOI for use in publications, etc., will always redirect to current release version (or devel if package is not in release yet)." class="external-link">10.18129/B9.bioc.BeadSorted.Saliva.EPIC</a></p>
</td>
</tr>
<tr class="odd">
<td>Brain</td>
<td>DLPFC (2012)</td>
<td>DLPFC</td>
<td>
<p>J Guintivano et al. (2013). A cell epigenotype specific model for
the correction of brain cellular heterogeneity bias and its application
to age, brain region and major depression. <em>Epigenetics</em>,
<em>8</em>(3), 290-302. doi: <a href="https://doi.org/10.4161/epi.23924" class="external-link">10.4161/epi.23924</a></p>
<pre><code><span><span class="va">FlowSorted.DLPFC.450k</span></span></code></pre>
<p><a href="https://doi.org/doi:10.18129/B9.bioc.FlowSorted.DLPFC.450k" title="DOI for use in publications, etc., will always redirect to current release version (or devel if package is not in release yet)." class="external-link">10.18129/B9.bioc.FlowSorted.DLPFC.450k</a></p>
</td>
</tr>
</tbody>
</table>
<p>The figure below focused on our optimization of the deconvolution
pipeline in the blood tissue. The four main steps of cellular
deconvolution are reference selection, data normalization, feature
selection, and regression. The function that calls each of the step is
bolded on the left, and the detail of each function is described in the
<a href="#usage">Usage</a> section.</p>
<p><embed src="CellsPickMe_pipeline.pdf"></embed>{fig.dim =” c(8, 6)“}</p>
</div>
</div>
<div class="section level2">
<h2 id="cell-type-deconvolution">Cell Type Deconvolution<a class="anchor" aria-label="anchor" href="#cell-type-deconvolution"></a>
</h2>
<p>In addition to improved methods for feature selection to optimize
prediction process, the package also includes several features to assess
prediction accuracy.</p>
<p>The <strong>CellsPickMe</strong> package includes 5 core
functions:</p>
<ul>
<li>
<code><a href="../reference/getRef.html">getRef()</a></code> finds and load the tissue-specific reference
data set</li>
<li>
<code><a href="../reference/combData.html">combData()</a></code>combines the reference and the user’s sample
data sets together</li>
<li>
<code><a href="../reference/pickProbes.html">pickProbes()</a></code> picks features that best distinguish cell
types (multiple feature selection methods available)</li>
<li>
<code>clusterScore()</code> assesses the performance of selected
features in clustering cell types in the reference</li>
<li>
<code><a href="../reference/predictCT.html">predictCT()</a></code> estimate cell type proportions based on
selected probes, with the ability to evaluate</li>
</ul>
<p>The <code><a href="../reference/pickProbes.html">pickProbes()</a></code> function can be computationally
intensive, depending on the methods for feature selection. Parallel
computing is available and highly recommended for improved efficiency.
Detailed usage is available <a href="CellsPickMe::pickProbes()">here</a></p>
<p>The following data are required for performing the cell type
proportion prediction with <strong>CellsPickMe</strong>: - DNAme data
(RGChannelSet Object is recommended, beta matrix is accepted as well) *
An appropriate reference can be obtained with the <code><a href="../reference/getRef.html">getRef()</a></code>
function. Otherwise the user can specify their own reference dataset as
well (needs to be a RGChannelSet Object with CellTypes information in
the pData)</p>
<hr>
<div class="section level3">
<h3 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h3>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu">install_github</span><span class="op">(</span><span class="st">"maggie-fu/CellsPickMe"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h3>
<p>The <strong>CellsPickMe</strong> package takes DNA methylation data
generated from Illumina microarray and predict its cellular composition
based on cell types available in the reference profiles. Currently, the
algorithm is compatible for peripheral blood, cord blood, saliva, and
brain (neuron vs non-neuron).</p>
<div class="section level4">
<h4 id="obtain-reference-dataset">Obtain reference dataset<a class="anchor" aria-label="anchor" href="#obtain-reference-dataset"></a>
</h4>
<p>From the available list of reference datasets (“Reinius”, “IDOL”,
“Extended”, “Cord”, “UniBlood7”, “UniBlood13”, “UniBlood19”, “DLPFC”,
and “Middleton”), select one that is appropriate for your sample (based
on tissue and sample age)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://maggie-fu.github.io/CellsPickMe/">CellsPickMe</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Request the UniBlood19 reference with no normalization</span></span>
<span><span class="va">ref_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/getRef.html">getRef</a></span><span class="op">(</span>ref <span class="op">=</span> <span class="st">"IDOL"</span>, normType <span class="op">=</span> <span class="st">"None"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="normalize-sample-and-reference-datasets-together">Normalize sample and reference datasets together<a class="anchor" aria-label="anchor" href="#normalize-sample-and-reference-datasets-together"></a>
</h4>
<p>Normalize user’s sample and reference data sets together to reduce
batch effect and improve prediction accuracy. Option for
<code>normType</code> includes “Noob”, “Funnorm”, “Quantile”,
“Quantile.b”, “None”, with the first 3 options being exclusively for
RGChannelSet objects, and Quantile.b for beta matrix.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Load example blood cell mixture, subsetted from the IDOL dataset (GSE110554)</span></span>
<span><span class="va">test_dat</span> <span class="op">&lt;-</span> <span class="fu">CellsPickMe</span><span class="fu">::</span><span class="va"><a href="../reference/IDOL_mixed_cells.html">IDOL_mixed_cells</a></span></span>
<span> </span>
<span><span class="co"># Combine sample and reference data sets together, followed by normalization (if selected)</span></span>
<span><span class="va">comb_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/combData.html">combData</a></span><span class="op">(</span>dataset <span class="op">=</span> <span class="va">test_dat</span>, </span>
<span>                     reference <span class="op">=</span> <span class="va">ref_dat</span><span class="op">$</span><span class="va">reference</span>, </span>
<span>                     cellTypes <span class="op">=</span> <span class="va">ref_dat</span><span class="op">$</span><span class="va">cellTypes</span>, </span>
<span>                     class <span class="op">=</span> <span class="st">"rgset"</span>,               <span class="co">#c("rgset", "betas")</span></span>
<span>                     normType <span class="op">=</span> <span class="st">"None"</span><span class="op">)</span>             <span class="co"># c("Noob", "Funnorm", "Quantile", "Quantile.b", "None")</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="pick-features-that-best-distinguish-cell-type">Pick features that best distinguish cell type<a class="anchor" aria-label="anchor" href="#pick-features-that-best-distinguish-cell-type"></a>
</h4>
<p>The <strong>CellsPickMe</strong> package supports feature selection
with either the traditional T-test, or with machine-learning-based
methods such as elastic net and random forest to obtain a curated list
of features that are highly predictive of cell types.</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="5%">
<col width="86%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>ID for the <code>caretMods</code> argument</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>least absolute shrinkage and selection operator</td>
<td>lasso</td>
<td>Use L1 regularization to optimize shrinkage of coefficients
uncorrelated with outcome of interest.</td>
</tr>
<tr class="even">
<td>elastic net</td>
<td>EN</td>
<td>Elastic Net is a linear regression-based method that can perform
feature selection by using L1 and L2 regularization.</td>
</tr>
<tr class="odd">
<td>random forest</td>
<td>RF</td>
<td>Random Forest is a decision tree-based ensemble method that can
perform feature selection by evaluating the importance of each feature
in the model. The Gini importance or Mean Decrease Impurity (MDI) are
two ways to calculate feature importance in Random Forest.</td>
</tr>
<tr class="even">
<td>boosted logistic regression</td>
<td>BLR</td>
<td>Boosted logistic regression is a machine learning algorithm that
combines logistic regression with boosting, a technique for iteratively
improving the performance of weak learners. # In boosted logistic
regression, a weak logistic regression model is trained on the data, and
then additional weak logistic regression models are trained to correct
the errors of the previous models. Each subsequent model focuses more on
the data points that were misclassified by the previous models, thus
improving the overall accuracy of the model.</td>
</tr>
<tr class="odd">
<td>classification and regression tree</td>
<td>CART</td>
<td>Decision Trees can perform feature selection by evaluating the
information gain or Gini impurity of each feature when deciding how to
split the data. The important features are those that lead to the
highest reduction in impurity or the highest information gain.</td>
</tr>
<tr class="even">
<td>gradient boosted machine</td>
<td>GBM</td>
<td>GBM is another ensemble-based method that can perform feature
selection by iteratively building trees that focus on predicting the
errors of previous trees. In this process, GBM can identify important
features that improve the model’s performance.</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Pick probes with T tests</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pickProbes.html">pickProbes</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                     probeList <span class="op">=</span> <span class="st">"Ttest"</span>, <span class="co">#c("Ttest", "Caret_CV", "Caret_LOOCV", "IDOL")</span></span>
<span>                     probeSelect <span class="op">=</span> <span class="st">"both"</span>, <span class="co">#c("both", "any")</span></span>
<span>                     nProbes <span class="op">=</span> <span class="fl">100</span>, <span class="co"># number of probes to pick for each cell type</span></span>
<span>                     p.val <span class="op">=</span> <span class="fl">0.05</span>,  <span class="co"># max pval</span></span>
<span>                     min.delta.beta <span class="op">=</span> <span class="fl">0.05</span>, <span class="co"># min delta beta</span></span>
<span>                     plotRef <span class="op">=</span> <span class="cn">T</span>, <span class="co"># plot heatmap?</span></span>
<span>                     verbose <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co">### Set up server for parallelization - run the code if picking probes with Caret</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/RevolutionAnalytics/doparallel" class="external-link">doParallel</a></span><span class="op">)</span></span>
<span><span class="va">cl</span> <span class="op">&lt;-</span> <span class="fu">makeCluster</span><span class="op">(</span><span class="fu">detectCores</span><span class="op">(</span><span class="op">)</span> <span class="op">-</span> <span class="fl">1</span><span class="op">)</span> <span class="co"># change as needed</span></span>
<span><span class="fu">registerDoParallel</span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Pick probes with repeated cross validation with lasso and elastic net</span></span>
<span><span class="va">probes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/pickProbes.html">pickProbes</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                     probeList <span class="op">=</span> <span class="st">"Caret_LOOCV"</span>, <span class="co">#c("Ttest", "Caret_CV", "Caret_LOOCV", "IDOL")</span></span>
<span>                     caretMods <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"lasso"</span>, <span class="st">"EN"</span><span class="op">)</span>,  <span class="co">#c("lasso", "EN", "BLR", "CART", "RF", "GBM")</span></span>
<span>                     filterK <span class="op">=</span> <span class="fl">1000</span>, <span class="co"># number of probes to put into the predictor for each cell type</span></span>
<span>                     seed <span class="op">=</span> <span class="fl">1</span>, </span>
<span>                     plotRef <span class="op">=</span> <span class="cn">F</span>, <span class="co"># plot heatmap?</span></span>
<span>                     verbose <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># FilterK results used as input for feature selection</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">$</span><span class="va">coefs</span><span class="op">$</span><span class="va">probeList</span><span class="op">$</span><span class="va">CD8T</span><span class="op">$</span><span class="va">tTestTopK</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Picked probes with EN and the estimated coefficients</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">$</span><span class="va">coefs</span><span class="op">$</span><span class="va">probeCoefs</span><span class="op">$</span><span class="va">EN</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Picked probes with EN for each cell type and the cross validation performance </span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">probes</span><span class="op">$</span><span class="va">coefs</span><span class="op">$</span><span class="va">probeList</span><span class="op">$</span><span class="va">CD8mem</span><span class="op">$</span><span class="va">EN</span><span class="op">$</span><span class="va">coefs</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="assess-clustering-stability">Assess clustering stability<a class="anchor" aria-label="anchor" href="#assess-clustering-stability"></a>
</h4>
<p>To further evaluate the performance of the selected probes, pvClust
can be applied to assess whether the picked probes can be used to
generate the correct cluster (cell type labeling) in reference data</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">clustAU</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/identClust.html">identClust</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>,</span>
<span>                      probes <span class="op">=</span> <span class="va">probes</span>,</span>
<span>                      parallel <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="estimate-cell-type-proportion">Estimate cell type proportion<a class="anchor" aria-label="anchor" href="#estimate-cell-type-proportion"></a>
</h4>
<p>Finally, the selected features are used to estimate cell type
proportions in the sample data set. We also incorporate the CETYGO score
(see Reference) to estimate prediction performance even in the absence
of a validation cohort / ground truth cell count.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/predictCT.html">predictCT</a></span><span class="op">(</span>dataNormed <span class="op">=</span> <span class="va">comb_dat</span>, </span>
<span>                 probes <span class="op">=</span> <span class="va">probes</span>, </span>
<span>                 method <span class="op">=</span> <span class="st">"CP"</span>,  <span class="co">#c("CP", "RPC", "SVR")</span></span>
<span>                 removenRBC <span class="op">=</span> <span class="cn">F</span>, <span class="co"># remove nRBC?</span></span>
<span>                 verbose <span class="op">=</span> <span class="cn">T</span>, </span>
<span>                 cetygo <span class="op">=</span> <span class="cn">T</span><span class="op">)</span> <span class="co"># CETYGO to assess reference appropriateness (RMSE evaluation)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="examine-the-output">Examine the output<a class="anchor" aria-label="anchor" href="#examine-the-output"></a>
</h4>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Estimated proportions based on IDOL reference, no normalization, EN feature selection, and constraint projection</span></span>
<span><span class="co"># CETYOGO score column shows the CETYGO score for each sample</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">out</span><span class="op">$</span><span class="va">EN</span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># Basic visualization of deconvolution output</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/hadley/reshape" class="external-link">reshape2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">plt</span> <span class="op">&lt;-</span> <span class="va">out</span><span class="op">$</span><span class="va">EN</span><span class="op">[</span>, <span class="va">ref_dat</span><span class="op">$</span><span class="va">cellTypes</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html" class="external-link">as.data.frame</a></span><span class="op">(</span><span class="va">.</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>    <span class="fu">reshape2</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html" class="external-link">melt</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">plt</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">variable</span>, <span class="va">value</span>, color <span class="op">=</span> <span class="va">variable</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">2</span>, position <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/position_jitter.html" class="external-link">position_jitter</a></span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>x <span class="op">=</span> <span class="st">"cell type"</span>, </span>
<span>         y <span class="op">=</span> <span class="st">"cell type proportion (%)"</span>, </span>
<span>         title <span class="op">=</span> <span class="st">"Blood cell type proportion"</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_manual.html" class="external-link">scale_color_manual</a></span><span class="op">(</span>values <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span>Bcell_cord <span class="op">=</span> <span class="st">"#9C9EDEFF"</span>, </span>
<span>                                  Bnv <span class="op">=</span> <span class="st">"#7375B5FF"</span>, </span>
<span>                                  Bmem <span class="op">=</span> <span class="st">"#4A5584FF"</span>, </span>
<span>                                  Bcell <span class="op">=</span> <span class="st">"#7375B5FF"</span>,</span>
<span>                                  CD4T_cord <span class="op">=</span> <span class="st">"#CEDB9CFF"</span>, </span>
<span>                                  CD4nv <span class="op">=</span> <span class="st">"#B5CF6BFF"</span>, </span>
<span>                                  CD4mem <span class="op">=</span> <span class="st">"#637939FF"</span>, </span>
<span>                                  Treg <span class="op">=</span> <span class="st">"#8CA252FF"</span>, </span>
<span>                                  CD4T <span class="op">=</span> <span class="st">"#B5CF6BFF"</span>,</span>
<span>                                  CD8T_cord <span class="op">=</span> <span class="st">"#E7CB94FF"</span>, </span>
<span>                                  CD8nv <span class="op">=</span> <span class="st">"#E7BA52FF"</span>, </span>
<span>                                  CD8mem <span class="op">=</span> <span class="st">"#8C6D31FF"</span>, </span>
<span>                                  CD8T <span class="op">=</span> <span class="st">"#E7BA52FF"</span>,</span>
<span>                                  NK_cord <span class="op">=</span> <span class="st">"#7BBCB0FF"</span>, </span>
<span>                                  NK <span class="op">=</span> <span class="st">"#3A7C89FF"</span>, </span>
<span>                                  Mono_cord <span class="op">=</span> <span class="st">"#F3CBD3FF"</span>, </span>
<span>                                  Mono <span class="op">=</span> <span class="st">"#707070"</span>,</span>
<span>                                  Gran_cord <span class="op">=</span> <span class="st">"#D39C83FF"</span>, </span>
<span>                                  Gran <span class="op">=</span> <span class="st">"#A65461FF"</span>, </span>
<span>                                  Neu <span class="op">=</span> <span class="st">"#A65461FF"</span>, </span>
<span>                                  Bas <span class="op">=</span> <span class="st">"#7B4173FF"</span>,</span>
<span>                                  Eos <span class="op">=</span> <span class="st">"#A55194FF"</span>, </span>
<span>                                  nRBC <span class="op">=</span> <span class="st">"#843C39FF"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, vjust <span class="op">=</span> <span class="fl">0.9</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="citation">Citation<a class="anchor" aria-label="anchor" href="#citation"></a>
</h3>
<p>The manuscript detailing CellsPickMe and its use is currently under
preparation. For more information about this please contact Maggie Fu at
<a href="mailto:maggie.fu@bcchr.ca" class="email">maggie.fu@bcchr.ca</a>.</p>
</div>
<div class="section level3">
<h3 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h3>
<p>Depending on the options you used, please consider citing the
following references as this package is built on their data / code /
papers.</p>
<ol style="list-style-type: decimal">
<li>
<p>DS Vellame et al. (2023). Uncertainty quantification of
reference-based cellular deconvolution algorithms. <em>Epigenetics</em>
18, 1: 2137659. doi: <a href="https://doi.org/10.1080/15592294.2022.2137659" class="external-link">10.1080/15592294.2022.2137659</a></p>
<ul>
<li>Please cite this paper if you set the parameter
<code>cetygo = T</code> for <code><a href="../reference/predictCT.html">predictCT()</a></code>
</li>
</ul>
</li>
<li>
<p>LE Reinius et al. (2012). Differential DNA methylation in
purified human blood cells: implications for cell lineage and studies on
disease susceptibility. <em>PloS one</em>. <em>7</em>(7), e41361. doi:
<a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "Reinius")</code>
</li>
</ul>
</li>
<li>
<p>DC Koestler et al. (2016). Improving cell mixture deconvolution
by identifying optimal DNA methylation libraries (IDOL). <em>BMC
bioinformatics</em>. 17, 120. doi: <a href="https://doi.org/10.1186/s12859-016-0943-7" class="external-link">10.1186/s12859-016-0943-7</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "IDOL")</code>
</li>
</ul>
</li>
<li>
<p>LA Salas et al. (2022) Enhanced cell deconvolution of peripheral
blood using DNA methylation for high-resolution immune profiling.
<em>Nat Commun</em>. 13, 761. doi: <a href="https://doi.org/10.1038/s41467-021-27864-7" class="external-link">10.1038/s41467-021-27864-7</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "Extended")</code>
</li>
</ul>
</li>
<li>
<p>K Gervin et al. (2019). Systematic evaluation and validation of
reference and library selection methods for deconvolution of cord blood
DNA methylation data. <em>Clinical epigenetics</em>. 11, 1-15. doi: <a href="https://doi.org/10.1186/s13148-019-0717-y" class="external-link">10.1186/s13148-019-0717-y</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "Cord")</code>
</li>
</ul>
</li>
<li>
<p>LY Middleton et al. (2022). Saliva cell type DNA methylation
reference panel for epidemiological studies in children.
<em>Epigenetics</em>, <em>17</em>(2), 161-177. doi: <a href="https://doi.org/10.1371/journal.pone.0041361" class="external-link">10.1371/journal.pone.0041361</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "Middleton")</code>
</li>
</ul>
</li>
<li>
<p>J Guintivano et al. (2013). A cell epigenotype specific model for
the correction of brain cellular heterogeneity bias and its application
to age, brain region and major depression. <em>Epigenetics</em>,
<em>8</em>(3), 290-302. doi: <a href="https://doi.org/10.4161/epi.23924" class="external-link">10.4161/epi.23924</a></p>
<ul>
<li>Please cite this paper if you used this reference dataset,
i.e. <code>getRef(ref = "DLPFC")</code>
</li>
</ul>
</li>
<li>
<p>TJ Triche, et al. (2013). Low-level processing of Illumina
Infinium DNA Methylation BeadArrays. <em>Nucleic Acids Res</em>. 41,
e90. doi: <a href="http://www.dx.doi.org/10.1093/nar/gkt090" class="external-link">10.1093/nar/gkt090</a>.</p>
<ul>
<li>Please cite this paper if you used this normalization method,
i.e. <code>getRef(normType = "Noob")</code> or
<code>combData(normType = "Noob")</code>
</li>
</ul>
</li>
<li>
<p>JP Fortin et al. (2014). Functional normalization of 450k
methylation array data improves replication in large cancer studies.
<em>Genome Biology</em> 15, 503. doi: <a href="http://www.dx.doi.org/10.1186/s13059-014-0503-2" class="external-link">10.1186/s13059-014-0503-2</a>.</p>
<ul>
<li>Please cite this paper if you used this normalization method,
i.e. <code>getRef(normType = "Funnorm")</code> or
<code>combData(normType = "Funnorm")</code>
</li>
</ul>
</li>
<li>
<p>N Touleimat and J Tost. (2012). <em>Complete pipeline for
Infinium Human Methylation 450K BeadChip data processing using subset
quantile normalization for accurate DNA methylation estimation.</em>
<em>Epigenomics</em> 4, 325-341. doi: <a href="https://doi.org/10.2217/epi.12.21" class="external-link">10.2217/epi.12.21</a></p>
<ul>
<li>Please cite this paper if you used this normalization method,
i.e. <code>getRef(normType = "Quantile")</code> or
<code>combData(normType = "Quantile")</code>
</li>
</ul>
</li>
</ol>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Maggie Fu.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
